src/app/heroes/pagination/pagination.component.ts

```diff
@@ -22,13 +23,34 @@ export class PaginationComponent implements OnChanges {
     totalHeroes.currentValue && this.generatePages();
   }
 
-  generatePages() {
-    const totalPages: number = Math.ceil(this.totalHeroes / this.pageSize);
+  totalPagesNumber() {
+    return Math.ceil(this.totalHeroes / this.pageSize);
+  }
 
+  generatePages() {
     this.pageList$ = Observable
-      .range(1, totalPages)
-      .reduce((total, item) => [...total, item], [])
-      .take(5);
+      .range(1, this.totalPagesNumber())
+      .reduce((total, item) => [...total, item], []);
+  }
+
+  goToPreviousPage() {
+    if ((this.currentPage$.value - 1) > 0) {
+      this.currentPage$.next(this.currentPage$.getValue() - 1);
```

Comment:
You use here `currentPage$.getValue()` but in the line above you use `currentPage$.value`, what's the difference?